subsys/pm/pm.c

```diff
@@ -131,6 +131,13 @@ bool pm_system_suspend(int32_t ticks)
 	} else {
 		const struct pm_state_info *info;
 
+#ifdef CONFIG_PM_LOW_LATENCY_EVENTS
+		/*
+		 * In this case CPU needs to be fully wake up before the event is
+		 * triggered. We need to find out first the ticks to the next event
+		 */
+		ticks = pm_policy_next_low_latency_event_ticks(ticks);
```

Comment:
The arguments and returned value of this function are unintuitive. I would consider refactoring this code to something like:

```
bool pm_system_suspend(int32_t kernel_ticks)
{
  ...
  low_latency_event_ticks = pm_policy_next_low_latency_event_ticks();
  ticks = next_ticks_from_two(kernel_ticks, low_latency_event_ticks);
#else /* CONFIG_PM_LOW_LATENCY_EVENTS */
  ticks = kernel_ticks;
  ...
```

Reply:
Also, this ticks calculation could be moved before the spinlock to avoid doing it twice.

Reply:
@hubertmis should be Ok now, please review the new rework